packages/compiler/emitter-framework/builders/array.ts

```diff
@@ -0,0 +1,38 @@
+import { compilerAssert } from "../../core/index.js";
+import { Placeholder } from "../placeholder.js";
+import { EmitEntity, EmitterResult } from "../types.js";
+
+export class ArrayBuilder<T> extends Array {
```

Comment:
Nit (feel free to won't fix): Consider putting the builders together in a single file like builders.ts or renaming the files to include builder. I think array.ts/object.ts/string.ts is likely to trip me up when using Go-to-file.